packages/expo-battery/src/Battery.ts

```diff
@@ -52,26 +108,65 @@ export async function getPowerStateAsync(): Promise<PowerState> {
+// @needsAudit
+/**
+ * Subscribe to the battery level change updates.
+ *
+ * On iOS devices, the event fires when the battery level drops one percent or more, but is only
+ * fired once per minute at maximum.
+ *
+ * On Android devices, the event fires only when significant changes happens, which is when the
+ * battery level drops below [`"android.intent.action.BATTERY_LOW"`](https://developer.android.com/reference/android/content/Intent#ACTION_BATTERY_LOW)
+ * or rises above [`"android.intent.action.BATTERY_OKAY"`](https://developer.android.com/reference/android/content/Intent#ACTION_BATTERY_OKAY)
+ * from a low battery level. See [here](https://developer.android.com/training/monitoring-device-state/battery-monitoring)
+ * to read more from the Android docs.
+ *
+ * On web, the event never fires.
+ * @param listener A callback that is invoked when battery level changes. The callback is provided a
+ * single argument that is an object with a `batteryLevel` key.
+ * @return A `Subscription` object on which you can call `remove()` to unsubscribe from the listener.s
+ */
+export function addBatteryLevelListener(
+  listener: (event: BatteryLevelEvent) => void
+): Subscription {
   return BatteryEventEmitter.addListener('Expo.batteryLevelDidChange', listener);
 }
 
-export function addBatteryStateListener(listener: BatteryStateUpdateListener): Subscription {
+// @needsAudit
+/**
+ * Subscribe to the battery state change updates to receive an object with a [`Battery.BatteryState`](#batterystate)
+ * enum value for whether the device is any of the four states.
+ *
+ * On web, the event never fires.
+ * @param listener A callback that is invoked when battery state changes. The callback is provided a
+ * single argument that is an object with a `batteryState` key.
+ * @return A `Subscription` object on which you can call `remove()` to unsubscribe from the listener.
+ */
+export function addBatteryStateListener(
+  listener: (event: BatteryStateEvent) => void
+): Subscription {
   return BatteryEventEmitter.addListener('Expo.batteryStateDidChange', listener);
 }
 
-export function addLowPowerModeListener(listener: PowerModeUpdateListener): Subscription {
+// @needsAudit
+/**
+ * Subscribe to Low Power Mode (iOS) or Power Saver Mode (Android) updates. The event fires whenever
+ * the power mode is toggled.
+ *
+ * On web, the event never fires.
+ * @param listener A callback that is invoked when Low Power Mode (iOS) or Power Saver Mode (Android)
+ * changes. The callback is provided a single argument that is an object with a `lowPowerMode` key.
+ * @return A `Subscription` object on which you can call `remove()` to unsubscribe from the listener.
+ */
+export function addLowPowerModeListener(listener: (event: PowerModeEvent) => void): Subscription {
   return BatteryEventEmitter.addListener('Expo.powerModeDidChange', listener);
 }
 
 export {
   BatteryLevelEvent,
-  BatteryLevelUpdateListener,
```

Comment:
Wouldn't it be a breaking change for people who import these listener types? 🤔

Reply:
Fair point here 👍 
Nevertheless I guess we can handle it by proper CHANGELOGing 😉 
@Simek, just document it as a breaking change entry in an appropriate CHANGELOG file 💪

Reply:
> Wouldn't it be a breaking change for people who import these listener types? 🤔

Yup, it might be, but those types were not documented before, and all what those types does were wrapping specific Event types in callback function, obfuscating the type chain a bit IMO.

I will updated the changelog as suggested. Unfortunately, due to types change this cannot be ported back to v41 docs.

Reply:
Thanks for the explanation — if it wasn't documented before, it's not a breaking change 😉 
Anyway, just in case, let's keep it in 🙂